provisioning/ci/synapse-auto-stop/azure.tf

```diff
@@ -0,0 +1,165 @@
+}
+
+resource "azurerm_automation_account" "synapse_automation_account" {
+  resource_group_name = var.resource_group
+  name                = "${var.resource_group}-Shared-automation"
+  sku_name            = "Basic"
+  location            = local.location
+}
+
+resource "azurerm_automation_runbook" "synapse_shutdown_automation_runbook" {
+  name                    = "StopSynapseServer"
+  automation_account_name = azurerm_automation_account.synapse_automation_account.name
+  log_progress            = "false"
+  log_verbose             = "false"
+  resource_group_name     = var.resource_group
+  location                = local.location
+  runbook_type            = "PowerShell"
+  content = templatefile("./runbooks/StopSynapseServer.ps1", {
+    connection_name     = azuread_application.automation_application.display_name,
+    resource_group      = var.resource_group,
+    synapse_server_name = var.synapse_server_name
+  })
+}
+
+resource "azurerm_automation_schedule" "synapse_shutdown_automation_schedule" {
+  name                    = "StopSynapseServer"
+  automation_account_name = azurerm_automation_account.synapse_automation_account.name
+  resource_group_name     = var.resource_group
+  frequency               = "Day"
+  timezone                = "Europe/Prague"
+}
+
+resource "azurerm_automation_job_schedule" "synapse_shutdown_automation_runbook_schedule" {
+  automation_account_name = azurerm_automation_account.synapse_automation_account.name
+  resource_group_name     = var.resource_group
+  runbook_name            = azurerm_automation_runbook.synapse_shutdown_automation_runbook.name
+  schedule_name           = azurerm_automation_schedule.synapse_shutdown_automation_schedule.name
+}
+
+resource "azurerm_automation_certificate" "synapse_shutdown_automation_certificate" {
+  automation_account_name = azurerm_automation_account.synapse_automation_account.name
+  base64                  = data.azurerm_key_vault_secret.automation_certificate.value
+  name                    = azuread_application.automation_application.display_name
+  resource_group_name     = var.resource_group
+}
+
+resource "azurerm_automation_connection" "synapse_shutdown_automation_connection" {
+  automation_account_name = azurerm_automation_account.synapse_automation_account.name
+  name                    = azuread_application.automation_application.display_name
+  resource_group_name     = var.resource_group
+  type                    = "AzureServicePrincipal"
+  values = {
+    ApplicationId         = azuread_application.automation_application.application_id
+    CertificateThumbprint = azurerm_automation_certificate.synapse_shutdown_automation_certificate.thumbprint
+    SubscriptionId        = data.azurerm_client_config.current.subscription_id
+    TenantId              = data.azurerm_client_config.current.tenant_id
+  }
+}
+
+resource "azurerm_key_vault" "synapse_shutdown_keyvault" {
```

Comment:
Nakonec se mi to povedlo udelat tak, ze certifikat se nemusi generovat rucne lokalne ale dela se to pres KV